docs/site/Dependency-injection.md

```diff
@@ -301,9 +301,118 @@ Error: Circular dependency detected:
   lead
 ```
 
+## Dependency injection for bindings with different scopes
+
+Contexts can form a chain and bindings can be registered at different levels.
+The binding scope controls not only how bound values are cached, but also how
+its dependencies are resolved.
+
+Let's take a look at the following example:
+
+![binding-scopes](../img/binding-scopes.png)
+
+The corresponding code is:
+
+```ts
+import {inject, Context, BindingScope} from '@loopback/context';
+import {RestBindings} from '@loopback/rest';
+
+interface Logger() {
+  log(message: string);
+}
+
+class PingController {
+  constructor(@inject('logger') private logger: Logger) {}
```

Comment:
when injecting the logger using same name `logger`, how would the resolver distinguish between controller's logger and service's logger? According to the context?

Reply:
Yes, it’s determined by the resolution context, the owner context, and the binding scope.